File: lib/features/aliment/domain/aliment.dart
import 'package:flutter/foundation.dart';
import 'package:live_vitalist/features/aliment_bank/domain/aliment_bank_state.dart';
import 'package:live_vitalist/features/aliment/domain/aliment_data.dart';

part 'temporary_aliment.dart';
part 'instanced_aliment.dart';

@immutable
sealed class Aliment {
  final double servingSize;
  final String unit;

  const Aliment({
    required this.servingSize,
    required this.unit,
  });

  Map<String, dynamic> toJson();
  AlimentData readDataRef(AlimentBankState bank);
  Aliment copyWith({double? servingSize, String? unit});
}

File: server/src/features/aliment/domain/Aliment.ts
import { AlimentData } from "./AlimentData";

export interface Aliment {
    servingSize: number;
    unit: string;
}

export interface TemporaryAliment extends Aliment {
    alimentData: AlimentData;
}

export interface InstancedAliment extends Aliment {
    alimentID: string;
}
